src/qililab/automatic_calibration/calibration_controller.py

```diff
@@ -0,0 +1,476 @@
+# Copyright 2023 Qilimanjaro Quantum Tech
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Automatic-calibration Controller module, which works with notebooks as nodes."""
+from datetime import datetime, timedelta
+
+import networkx as nx
+
+from qililab.automatic_calibration.calibration_node import CalibrationNode
+from qililab.data_management import build_platform, save_platform
+from qililab.platform.platform import Platform
+
+
+class CalibrationController:
+    """Class that controls the automatic calibration sequence.
+
+    Using this class, you normally would run:
+        - ``CalibrationController.run_automatic_calibration()`` for calibrating the full graph.
+        - ``CalibrationController.maintain(node)`` for targeting and making sure a node works.
+
+    |
+
+    **The calibration workflow, has three levels of methods that manage it:**
+
+    The highest level method to apply, is ``run_automatic_calibration()`` which finds all the final nodes (don't have others past them), and runs ``CalibrationController.maintain()`` on those.
+
+    The two mid-level methods are ``maintain()`` and ``diagnose()``, the first one goes back to all the nodes dependencies starts, and working
```

Comment:
You were totally right, I manually modify it first, and then passed it through chatGPT, and much more clear now! 

Tell me if you still think otherwise please! 🙌